### DISCUSSION TITLE
Getting 413 error and CORS policy error at the same time (bug?)

### DISCUSSION THREAD

**Discussion by Vadorequest** [vercel/vercel#8175] [Help] [answered]
I'm trying to send a request to my Next.js API but it fails with several issues:
- The response status code is 413 ([Payload too large, 4.5MB max](https://vercel.com/docs/concepts/limits/overview#serverless-function-payload-size-limit)), although the "content-length" is only 1MB (1099528 is ~1MB, right?)
- The fetch request is blocked due to CORS policy, although I do allow CORS for that particular endpoint (and it works when the payload is smaller)

So, I believe there is an issue, mostly likely the 413 is sent and because of that my code isn't reached, and so my `res.setHeader('Access-Control-Allow-Origin', '*');` isn't applied, which result in a CORS issue.

I'm confused about the size, though. 1099528 bytes should be only 1MB, right? It's way below the allowed limit, it shouldn't return a 413 in the first place.

**Comment by Vadorequest** [comment, marked as answer]
Somehow, I'm right (or right-ish). I've deleted my "per-route" CORS headers, and added them to next.config.js instead, ~~and it just worked.~~

Edit: I also had to [increase the body size limit](https://nextjs.org/docs/api-routes/api-middlewares#custom-config).

![image](https://user-images.githubusercontent.com/3807458/179419505-046e1cf8-9379-4ef2-9e88-89c72d42eaa6.png)
![image](https://user-images.githubusercontent.com/3807458/179419511-14ce3b7c-a837-45f8-b2a7-0b334c2cbde5.png)
![image](https://user-images.githubusercontent.com/3807458/179420049-b130235e-edc9-4f74-8856-c547c3c14859.png)

**Reply by Vadorequest** [reply]
Well, now I'm getting `Body exceeded 1mb limit` in the response payload. It was supposed to be 4.5MB 🤔

**Reply by Vadorequest** [reply]
I assumed the 4.5Mb limit [I read there](https://vercel.com/docs/concepts/limits/overview#serverless-function-payload-size-limit) was what I was looking for, but it looks like the body has [a different config](https://nextjs.org/docs/api-routes/api-middlewares#custom-config) and can be increased.

**Reply by samcx** [reply]
@Vadorequest Is it possible to create a minimal reproduction of what you were testing above?
